src/lex.rs

```diff
@@ -39,18 +39,42 @@ impl Operator {
     }
 }
 
+struct NFunctionArgs {
+    args: Vec<f64>,
+}
+
+impl NFunctionArgs {
+    fn new(arity: usize, args: Vec<f64>) -> NFunctionArgs {
+        if arity != args.len() {
+            panic!("Unexpected, arity != args.len()");
+        }
+        NFunctionArgs { args }
+    }
+    fn get_n(&self, n: usize) -> f64 {
+        *self.args.get(n).unwrap()
+    }
+}
+
+macro_rules! arity_1 {
+    (|$a:ident| $b:expr) => { |n: NFunctionArgs| -> f64 { let $a = n.get_n(0); $b } }
+}
+macro_rules! arity_2 {
+    (|$a:ident, $b:ident| $c:expr) => { |n: NFunctionArgs| -> f64 { let ($a, $b) = (n.get_n(0), n.get_n(1)); $c } }
+}
+
 #[derive(Debug, Clone, PartialEq)]
 pub struct Function {
-    token: String,
-    relation: fn(f64) -> f64,
+    pub token: String,
+    relation: fn(NFunctionArgs) -> f64,
```

Comment:
I wonder if there is a better way to do this? I am thinking that it would be good if we can make use of const generics and do `[f64; N]` rather than storing it separately?

Reply:
Ohh nice, that's what I was initially looking for, thanks for the suggestion!:)